src/GovUk.Education.ExploreEducationStatistics.Common/Services/PostgreSqlRepository.cs

```diff
@@ -0,0 +1,99 @@
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Requests;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+using NpgsqlTypes;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Services;
+
+public class PostgreSqlRepository : IPostgreSqlRepository
+{
+    public async Task<TResponse?> GetJsonbFromPath<TDbContext, TRowId, TResponse>(
+        TDbContext context,
+        JsonbPathRequest<TRowId> request,
+        CancellationToken cancellationToken = default)
+        where TDbContext : DbContext
+        where TResponse : class
+    {
+        var jsonPathParam = new NpgsqlParameter("jsonPath", request.PathSegments);
+        var rowIdParam = new NpgsqlParameter("rowId", request.RowId);
+
+        var response = await context
+            .Set<JsonFragment>()
+            .FromSqlRaw(
+                sql: $"""
+                      SELECT "{request.JsonbColumnName}"#>@jsonPath "JsonString" FROM "{request.TableName}"
+                      WHERE "{request.IdColumnName}" = @rowId
+                      """,
+                parameters: [jsonPathParam, rowIdParam])
+            .SingleOrDefaultAsync(cancellationToken);
+
+        return response is not null
+               && response.JsonString is not null
+            ? JsonSerializer.Deserialize<TResponse>(response.JsonString)
+            : null;
+    }
+
+#pragma warning disable EF1002
+    public async Task<TValue?> SetJsonbAtPath<TDbContext, TValue, TRowId>(
```

Comment:
this could potentially be `private` for now. But I can definitely see a case where you would use it. Happy for you to leave it as either. Just wanted to point it out :)

Reply:
Yup thanks - gonna leave it public for the very reason that it could be useful for the future, ta!

Reply:
sounds good!